File: generators/python-bots/templates/python/bots/request-reply/listeners/elements_listener_impl.py
import logging
from sym_api_client_python.clients.sym_bot_client import SymBotClient
from sym_api_client_python.listeners.elements_listener import ElementsActionListener


class ElementsListenerImpl(ElementsActionListener):
    def __init__(self, sym_bot_client):
        self.bot_client = sym_bot_client

    async def on_elements_action(self, action):
        logging.debug('Elements Action Received', action)

File: generators/java-bots/templates/java/bots/elements/src/main/java/ElementsListenerImpl.java
import clients.SymBotClient;
import java.util.Map;
import listeners.ElementsListener;
import model.OutboundMessage;
import model.User;
import model.events.SymphonyElementsAction;

public class ElementsListenerImpl implements ElementsListener {
    private SymBotClient botClient;

    public ElementsListenerImpl(SymBotClient botClient) {
        this.botClient = botClient;
    }

    public void onElementsAction(User initiator, SymphonyElementsAction action) {
        Map<String, Object> formValues = action.getFormValues();
        botClient.getMessagesClient().sendMessage(action.getStreamId(), new OutboundMessage("Your action has been processed"));
        botClient.getMessagesClient().sendMessage(action.getStreamId(), new OutboundMessage(formValues.toString()));
    }
}
